packages/jellyfish-testing/src/fixture.ts

```diff
@@ -0,0 +1,54 @@
+import { Testing } from './index'
+import { poolpair } from '@defichain/jellyfish-api-core'
+
+/**
+ * TestingFixture setup complex fixtures for jellyfish testing.
+ */
+export class TestingFixture {
+  constructor (
+    private readonly testing: Testing
+  ) {
+  }
+
+  async createPoolPair (options: TestingPoolPairCreateRequest): Promise<poolpair.PoolPairInfo> {
+    if (options.a.symbol !== 'DFI') {
```

Comment:
Is that better if you have `await this.testing.generate(1)` for every calls?

Reply:
hmm.. seems its intention, is tit better manageable for multi-nodes??

Reply:
Yeah, it is intentional as block generation is quite expensive time-wise. (You can't generate too far into the future)

By design `fixture.ts` should only generate blocks they need, so we don't waste expensive testing cycle. As this is a shared fixture for the entirety of the jellyfish testing ecosystem.